src/main/java/io/wispforest/owo/serialization/SequenceDeserializer.java

```diff
@@ -0,0 +1,12 @@
+package io.wispforest.owo.serialization;
+
+import java.util.Iterator;
+
+public interface SequenceDeserializer<E> extends Iterator<E> {
+
+    int size();
+
+    boolean hasNext();
+
+    E next();
```

Comment:
Doesn't Iterator have these methods anyways? 🤨

Reply:
Yes

Reply:
Did such as a note to what is being implemented exactly and needs to be implemented